lib/ap/plan9/rename.c

```diff
@@ -43,15 +43,15 @@ rename(const char *from, const char *to)
 	f = f? f+1 : (char *)from;
 	t = t? t+1 : (char *)to;
 	n = 0;
-	if(f-from==t-to && strncmp(from, to, f-from)==0){
-		/* from and to are in same directory (we miss some cases) */
+	/* if(f-from==t-to && strncmp(from, to, f-from)==0){
```

Comment:
nasty. Hard to follow. Would be nicer if you just did this:
if (0 && (f-from == etc. etc.)
Then the code would compile, always, and we could come back and fix it later.